### PR TITLE
add min="0" to some parameters

### PR DESCRIPTION
**Pull request by bgruening**
as follow up from https://github.com/galaxyproject/tools-iuc/pull/288

### CODE REVIEW

File: tools/hisat2/hisat2.xml [hunk 1]
```diff
@@ -134,10 +134,10 @@
...
-                <param argument="-5" default="0" help="Trim N bases from 5' (left) end of each read before alignment" label="Trim 5' end" name="trim_five" type="integer" value="0" />
-                <param argument="-3" default="0" help="Trim N bases from 3' (right) end of each read before alignment" label="Trim 3' end" name="trim_three" type="integer" value="0" />
+                <param argument="-s" default="0" min="0" label="Skip the first N reads or pairs in the input" name="skip" type="integer" value="0" />
+                <param argument="-u" default="0" min="0" help="Align the first N reads or read pairs from the input (after the first N reads or pairs have been skipped), then stop." label="Stop after aligning N reads" name="stop_after" type="integer" value="0" />
+                <param argument="-5" default="0" min="0" help="Trim N bases from 5' (left) end of each read before alignment" label="Trim 5' end" name="trim_five" type="integer" value="0" />
+                <param argument="-3" default="0" min="0" help="Trim N bases from 3' (right) end of each read before alignment" label="Trim 3' end" name="trim_three" type="integer" value="0" />
```
Comment (nsoranzo) [inline tools/hisat2/hisat2.xml]: `default="0"` is not a recognized attribute, and there is already `value="0"` at the end of each line.

These should be removed also in other places in the XML file.

Comment (bgruening) [inline tools/hisat2/hisat2.xml]: Yes working on it.

Comment (bgruening) [inline tools/hisat2/hisat2.xml]: @nsoranzo ping

**Comment by bgruening** [comment]
I found some other bugs in the data manager and included fixes as well.

File: data_managers/data_manager_hisat2_index_builder/data_manager/hisat2_index_builder.xml [hunk 2]
```diff
@@ -8,13 +8,13 @@
         <exit_code range="1:" />
     </stdio>
     <command><![CDATA[
-        #if $gtf_input:
+        #if $advanced.adv_param_select == 'yes' and $advanced.gtf_input:
             ln -s "${gtf_input}" gtf_file.gtf &&
```
Comment (nsoranzo) [inline data_managers/data_manager_hisat2_index_builder/data_manager/hisat2_index_builder.xml]: Also here should be `${advanced.gtf_input}`

File: data_managers/data_manager_hisat2_index_builder/data_manager/hisat2_index_builder.xml [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
             python \$HISAT2_ROOT_DIR/bin/extract_splice_sites.py gtf_file.gtf > splice_sites.txt &&
             python \$HISAT2_ROOT_DIR/bin/extract_exons.py gtf_file.gtf > exon.txt &&
             ls -lh &&
         #end if
-        #if $snps:
+        #if $advanced.adv_param_select == 'yes' and $advanced.snps:
             ln -s "${all_fasta_source.fields.path}" genome.fa &&
             ln -s "${snps}" snps.tabular &&
```
Comment (nsoranzo) [inline data_managers/data_manager_hisat2_index_builder/data_manager/hisat2_index_builder.xml]: Also here should be `${advanced.snps}`

Comment (bgruening) [inline data_managers/data_manager_hisat2_index_builder/data_manager/hisat2_index_builder.xml]: Nice, saw this also and fixed it already. Wonder if Dave has run this on real data.

**Comment by nsoranzo** [comment]
:+1:

**Comment by bgruening** [comment]
Awesome! Thanks!

**Pull request merged**